exceptions/exceptions.hh

```diff
@@ -348,7 +348,9 @@ public:
 // It is advised to use this mechanism mainly for exceptions which can
 // happen frequently, e.g. signalling timeouts, overloads or rate limits.
 using coordinator_exception_container = utils::exception_container<
-    mutation_write_timeout_exception
+    mutation_write_timeout_exception,
+    read_timeout_exception,
+    read_failure_exception
```

Comment:
For the future, we can consider creating an std::error_category subclass for scylla errors so we can drop the variant (and instead use switches on the error code). But let's delay that until we complete the conversions, it's less important than converting the full stack.

Reply:
Do error codes support additional data apart from the error code itself? For example, `mutation_write_timeout_exception` has multiple fields which includes keyspace name, table name, consistency level, etc. - basically all information needed for the CQL layer to send a WRITE_TIMEOUT response to the client ([link](https://github.com/scylladb/scylla/blob/10880fb0a720e3dcddcb301f8f1c624923e89a03/exceptions/exceptions.hh#L152) to the definition in the code). If they don't, then I'm not sure how we could propagate this information.

Perhaps the situation is better on the replica side... If so, then maybe we could use error codes there.

Reply:
We can have class scylla_error : public std::system_error { extra fields } to store stuff. Or even class scylla_error : public std::exception { enum code, extra fields } and drop the error_category idea, which is only useful if we mix multiple categories (like syscall / dns / whatever error categories, which we don't).